src/modules/campath/camio.rs

```diff
@@ -0,0 +1,143 @@
+use std::array::from_fn;
+
+use nom::bytes::complete::{tag, take_until};
+use nom::character::complete::{multispace0, newline};
+use nom::combinator::{all_consuming, map, opt};
+use nom::multi::separated_list0;
+use nom::sequence::{delimited, preceded, tuple};
+use nom::IResult;
+
+use super::common::{angle_diff, cam_float, lerp, ViewInfo};
+
+#[derive(Clone, Copy)]
+pub struct ViewInfoCamIO {
+    pub viewinfo: ViewInfo,
+    pub time: f64,
+    pub fov: f64,
+}
+
+#[derive(Clone)]
+pub struct CamIO {
+    pub campaths: Vec<ViewInfoCamIO>,
+}
+
+impl CamIO {
+    fn find_next_entry(&self, time: f64) -> usize {
+        if time <= 0. || self.campaths.is_empty() {
+            return 1;
+        }
+
+        let end = self.campaths.last().unwrap();
+
+        if time >= end.time {
+            return self.campaths.len();
+        }
+
+        // TODO: it can be more efficient
+        self.campaths
+            .iter()
+            .enumerate()
+            .find(|(_, e)| time < e.time)
```

Comment:
.position()

also you can combine it with condition above using .unwrap_or(self.campaths.len())